🐛 Error when building documentation with Python 3.13+ (`No module named 'imghdr'`)

## Description

❌ Since [Feb 17](https://github.com/nightscout/nightscout.github.io/actions/runs/13372948309/job/37345508906#step:5:11), the `documentation.yaml` workflow has been failing repeatedly with this error:

`Could not import extension sphinx.builders.epub3 (exception: No module named 'imghdr'`

```
Run sphinx-build -b dirhtml docs _build
sphinx-build -b dirhtml docs _build
shell: /usr/bin/bash -e {0}
env:
  pythonLocation: /opt/hostedtoolcache/Python/3.13.2/x64
  LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.13.2/x64/lib
Running Sphinx v5.3.0

Extension error:
Could not import extension sphinx.builders.epub3 (exception: No module named 'imghdr')
Error: Process completed with exit code 2.
```


## Cause

The **`imghdr`** module has been [removed from the Python Standard Library as of Python 3.13](https://github.com/sphinx-doc/sphinx/issues/10440), following its deprecation in Python 3.11.

The `documentation.yaml` GitHub workflow highlighted this bug as it uses Python v3.13+.
I reproduced the bug **locally** after upgrading Python to version 3.13 and attempting to build the documentation.
Therefore, this is not a CI/CD bug.  

➡️ The cause is the outdated package(s) using the ìmghdr` module that no longer exists as of Python 3.13+.

```
cd nighscout.github.io

pyenv install 3.13.2
pyenv local 3.13.2

python  -m venv venv
pip install -r requirements.txt

rm -rf _build
sphinx-build -M dirhtml docs/ _build/

#=> Running Sphinx v5.3.0
#=>
#=> Extension error:
#=> Could not import extension sphinx.builders.epub3 (exception: No module named 'imghdr')
```


## Solution

A quick search led me to conclude that `sphinx` version 5.3 is the only package that uses it.


```shell
grep -r imghdr venv/lib/python3.13/site-packages/

Binary file venv/lib/python3.13/site-packages//sphinx/util/__pycache__/images.cpython-313.pyc matches
venv/lib/python3.13/site-packages//sphinx/util/images.py:import imghdr
venv/lib/python3.13/site-packages//sphinx/util/images.py:    imgtype = imghdr.what(stream)
venv/lib/python3.13/site-packages//sphinx/util/images.py:    """An additional imghdr library helper; test the header is SVG's or not."""
venv/lib/python3.13/site-packages//sphinx/util/images.py:# install test_svg() to imghdr
venv/lib/python3.13/site-packages//sphinx/util/images.py:# refs: https://docs.python.org/3.6/library/imghdr.html#imghdr.tests
venv/lib/python3.13/site-packages//sphinx/util/images.py:imghdr.tests.append(test_svg)
```

🎯 So, I suggest updating the project packages/dependencies:
- using `imghdr`, i.e. `sphinx`
- but also all the other direct [dependencies/packages](https://github.com/nightscout/nightscout.github.io/blob/cd61404848dc77f93932858848b89820ec499a83/requirements.txt) in one go.

--- a/.github/workflows/documentation.yaml
+++ b/.github/workflows/documentation.yaml
@@ -8,6 +8,8 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
+        with:
+          python-version: 3.13
       - name: Install dependencies
         run: |
           pip install -r requirements.txt

--- a/.github/workflows/documentation.yaml
+++ b/.github/workflows/documentation.yaml
@@ -6,8 +6,8 @@ jobs:
   docs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
           pip install -r requirements.txt
@@ -15,7 +15,7 @@ jobs:
         run: |
           sphinx-build -b dirhtml docs _build
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+        uses: peaceiris/actions-gh-pages@v4
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/source' }}
         with:
           publish_branch: gh-pages

--- a/.readthedocs.yaml
+++ b/.readthedocs.yaml
@@ -7,9 +7,9 @@ version: 2
 
 # Set the version of Python and other tools you might need
 build:
-  os: ubuntu-22.04
+  os: ubuntu-24.04
   tools:
-    python: "3.11"
+    python: "3.13"
     # You can also specify other tool versions:
     # nodejs: "19"
     # rust: "1.64"

--- a/.readthedocs.yaml
+++ b/.readthedocs.yaml
@@ -17,6 +17,7 @@ build:
 
 # Build documentation in the docs/ directory with Sphinx
 sphinx:
+  builder: dirhtml
    configuration: docs/conf.py
 
 # If using Sphinx, optionally build your docs in additional formats such as PDF

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
-Sphinx==5.3.0
+sphinx==8.2.0
 sphinx-copybutton==0.5.2
-sphinx_design==0.4.1
-sphinx-rtd-theme==1.2.0
-myst-parser==1.0.0
-sphinx-notfound-page==1.0.4
+sphinx_design==0.6.1
+sphinx-rtd-theme==3.0.2
+myst-parser==4.0.1
+sphinx-notfound-page==1.1.0

